src/services/admin/admin-router.ts

```diff
@@ -0,0 +1,48 @@
+import { Router } from "express";
+import { StatusCodes } from "http-status-codes";
+import { Database } from "../../database";
+import RoleChecker from "../../middleware/role-checker";
+import { Role } from "../auth/auth-models";
+import dotenv from "dotenv";
+import { validateQrHash } from "./admin-utils";
+
+dotenv.config();
+
+const adminRouter = Router();
+
+adminRouter.post(
+    "/scan/",
+    RoleChecker([Role.Enum.ADMIN]),
+    async (req, res, next) => {
+        try {
+            const { qrCode } = req.body;
+            if (!qrCode) {
+                console.log("made it to if");
+                return res
+                    .status(StatusCodes.BAD_REQUEST)
+                    .json({ error: "QR code is required" });
+            }
+            const { userId, expTime } = validateQrHash(qrCode);
```

Comment:
does the QrHash not contain an eventId as well?

Reply:
or well, our request should contain an eventId and qrCode, can you make a validator in admin service with these 2 fields and verify off that?